[Robin] 미션3 : 보드 초기화 하기 PR

malaheaven opened the pull request:
안녕하세요, Robin입니다. 미션3 보드 초기화하기 리뷰 요청 드립니다. 😄 
먼저, 제 코드를 리뷰해주시는 리뷰어분에게 감사를 드립니다.

커밋 로그를 더 잘게 잘게 나눠서 올려야 하는데, 막상 개발할 때는 생각이 안 나다가 꼭 PR 요청할 때가 되어서야 보이는 제 commit이 아쉽게 느껴지네요. 다음번엔 더욱 발전된 commit 로그를 작성할 수 있도록 노력해야겠습니다.

덧, 사실대로 양심 고백할 것이 하나 있는데, Board 클래스 중복 리팩토링을 진행하다 코드 중복 제거에 영.. 감이 잡히지 않아, 참고 소스의 BoardTest 코드만(테스트 코드를) 보고 진행했습니다. 최대한 보지 않으려고 했는데, 그만 판도라의 상자를 열어버렸네요. 😢 

[미션 진행 목록]
1. PawnTest에 요구사항 코드 추가 
2. Pawn 클래스
   1. Pawn 색에 따른 표시 문자 상수로 부여
      * whitePawn -> 'p'
      * blackPawn -> 'P'
   2. Pawn 생성자 public Pawn(String color, char representation) 로 변경 (기존: public Pawn(String color))
3. BoardTest 요구사항 코드 적용
4. Board Class 
   1. 흰색 Pawn과 검은색 Pawn을 생성해 저장하는 List를 추가
      * List<Pawn> pawn = new ArrayList<>();
   2. whitePawnsAdd(Pawn pawn): 흰색 폰 리스트에 add 해주는 메소드
   3. whitePawnsAdd(Pawn pawn): 검은색 폰 리스트에 add 해주는 메소드
   4. findWhitePawn(int index): 흰색 폰 리스트에서 인덱스 번호로 리스트 안에 들어있는 원소를 찾을 수 있는 메소드
   5. findBlackPawn(int index): 검은색 폰 리스트에서 인덱스 번호로 리스트 안에 들어있는 원소를 찾을 수 있는 메소드
   6. size(): Board에 추가되는 기물 size를 구하는 메소드
   7. initialize(): 흰색 Pawn 열에 보드 사이즈 만큼 pppppppp 추가 검은색 Pawn 열에 PPPPPPPP 추가
   8. getWhitePawnsResult(): 흰색 Pawn 열의 값이 return 될 수 있도록 적용
   9. getBlackPawnsResult(): 검정색 Pawn 열의 값이 return 될 수 있도록 적용
   10. print(): 체스판의 현재 상태를 콘솔 화면에 출력이 가능하게 해주는 메소드 
5. GameMain 클래스 구현
   1. main()메소드에 체스게임을 시작하고 종료하는 기능을 추가
      * `start` 명령을 입력하면 체스게임을 시작하고 체스판의 현재 상태를 콘솔 화면에 출력
      * `end` 명령을 입력하면 체스 게임을 종료하고 `Bye~` 를 출력
   
      <출력 결과>
![image](https://user-images.githubusercontent.com/45054467/108332643-60dd6800-7213-11eb-8a7f-636e69463808.png)

malaheaven committed:
Mission2 pr 코멘트  적용

Pawn 클래스 this()로 기존에 있는 다른 생성자 호출해서 활용
테스트 클래스와 테스트 메소드 제어자 생략 및 test 어노테이션 붙지 않는 메소드 private 추가
기존에 테스트 클래스에서 assertEqual로 되어있던 부분 assertThat으로 변경

malaheaven committed:
요구사항 PawnTest 추가 및 개선

요구사항에 있는 PawnTest 추가하고 assertEquals로 작성되어있던 코드 assertThat으로 변경
DisplayName어노테이션을 이용하여 해당 단위테스트 내용 표시

malaheaven committed:
Pawn 색에 따른 출력 문자 부여(PawnTest 요구사항에 맞춰 Pawn 클래스 내용 추가)

흰색과 검정색 표시 문자 상수 추가(흰색은 p, 검은색은 P)
기본 생성자는 흰색과 흰색 표시문자 p 를 값으로 가진다. 이는 this()를 이용하여 다른 생성자로 보냄
표시문자를 리턴 받고 싶을 때를 위한 getRepresentation() 메소드 추가

malaheaven committed:
BoardTest 요구사항 적용

체스판 초기화 메소드에 관한 테스트
initialize() 메소드 테스트 내용 적용

malaheaven committed:
Board Class initialize() getWhitePawnsResult() getBlackPawnsResult() 메소드 추가

initialize() 메소드에 흰색 검은색 Pawn을 생성해 저장하는 List 추가
initialize() 메소드에 흰색 검은색 리스트에 해당 Pawn 객체를 넣어주는 코드 추가
StringBuilder에 각각 흰색 검정색에 대한 representation 문자열 내용 더하는 코드 추가
getWhitePawnsResult() 흰색 Pawn 열의 값이 return 될 수 있도록 적용
getBlackPawnsResult() 검정색 Pawn 열의 값이 return 될 수 있도록 적용

malaheaven committed:
Pawn클래스 Pawn(String color) 생성자 삭제

- PawnTest 에서 Pawn(String color)생성자에 대한 내용이 없어, 생성자 삭제

malaheaven committed:
BoardTest Create()메소드 개선 및 assertEquals -> assertThat으로 개선

- BoardTest의 Create()메소드는 Pawn클래스의 Pawn(String color) 생성자를 삭제후 PawnTest가 잘 테스트가 통과하는지 확인을 위해 코드를 실행했지만, BoardTest에서 기존 생성자로 구현되어있는 부분 때문에 PawnTest가 코드가 실행되지 않아 Pawn(String color, char representation) 생성자로 맞춰 코드 수정
- assertEquals로 구현되어있는 부분 assertThat으로 개선

malaheaven committed:
Board 클래스 리팩토링 진행(whitePawnsAdd,blackPawnsAdd,findWhitePawn,findBlackPawn 추가)

기존 Mission2에서 작성되었던 메소드들 처럼 whitePawns, blackPawns 리스트들의 메소드를 추가하라는 뜻으로 미션을 이해해 각각의 add()메소드, find()메소드를 추가
최대한 중복코드를 제거하고 메소드가 한 가지 일만 처리하도록 구현.
- whitePawns,blackPawns 리스트 생성 및 각각의 Add(),findWhitePawn() 및 findBlackPawn()  메소드 추가
- size() 메소드 같은 경우는 board에 몇 개의 기물이 생성되어있는지 값을 구하라는 뜻으로 미션이 이해되어, board에 추가된 기물 값을 리턴하도록 개선
- initialize() -> board판 초기화 메소드 같은 경우는 지역변수로 boardSize를 8로 초기화해 boardSize 만큼 흰색 폰, 검정색 폰을 리스트에 추가하도록 함
- getWhitePawnsResult(), getBlackPawnsResult()의 경우 각각의 StringBuilder를 통해 List의 값 합쳐서 toString으로 반환

malaheaven committed:
BoardTest 리팩토링 진행 및 코드 추가

Board클래스 리팩토링 진행으로 Test 코드 변화가 필요하여 진행함.

malaheaven committed:
Board 클래스 ArrayList<Pawn> pawns = new ArrayList<Pawn> 로 되어있는 부분 List<Pawn> pawns = new ArrayList<List>로 변경

ArrayList로 되어있던 부분 List의 기능까지 사용할 수 있도록 변경(자바의 다형성의 특징을 상속받아 사용)

malaheaven committed:
Board클래스 print() 메소드 구현

Board의 initialize() 메소드를 통해 요구사항과 같이 출력되게 하는 메소드 추가
출력할 결과를 StringBuilder에 저장한 후 String으로 반환하는 메소드

malaheaven committed:
BoardTest 보드 초기화가 잘 print 되는지 확인하는 print() 메소드 생성

malaheaven committed:
GameMain 클래스 구현

main()메소드에 체스게임을 시작하고 종료하는 기능을 추가
start 명령을 입력하면 체스게임을 시작하고 체스판의 현재 상태를 콘솔 화면에 출력
end 명령을 입력하면 체스 게임을 종료하고 Bye~를 출력

malaheaven committed:
ReadMe에 Mission3 구현 사항 작성

ksundong commented on README.md:
````diff
@@ -27,3 +27,41 @@
+      ........
+      pppppppp
+      ........
+      ```
````
요구사항 정리하시는거 좋네요. 😊

ksundong commented on src/main/java/net/honux/chess/Board.java:
```diff
@@ -3,20 +3,82 @@
 public class Board {
 
-    ArrayList<Pawn> pawns = new ArrayList<Pawn>();
+    List<Pawn> whitePawns = new ArrayList<Pawn>();
+    List<Pawn> blackPawns = new ArrayList<Pawn>();
```
접근 제어자에 유의해줍시다. 꼭 학습하고 넘어가주세요!

malaheaven replied:
넵넵 클래스 내부에서만 사용하는 거라, private를 붙여줘야 하는 데, default로 했었네요. 꼭 유의하겠습니다.

ksundong commented on src/main/java/net/honux/chess/Board.java:
```diff
@@ -3,20 +3,82 @@
-    public void add(Pawn pawn) {
-        pawns.add(pawn);
+
+    public void whitePawnsAdd(Pawn pawn){
+        whitePawns.add(pawn);
+    }
```
`addWhitePawn`이 저는 더 적당한 것 같아요.
그리고, 여기에 정말 White Pawn만 들어올 수 있는거 확실한가요? 👀

malaheaven replied:
넵넵 변경했습니다.
그리고 해당 메소드에서 다른 Pawn 색상이 들어오면 오류를 내는 것으로 변경했어요.
이렇게 하는게 좋을까? 라고 생각하면 그리 자신은 없네요. 
차라리 색을 구분해주는 메소드를 하나 만드는 게 나을지.. 아닐지.. 우선 이렇게 진행했습니다

ksundong replied:
일단 현재단계에선 충분하다고 생각합니다.

ksundong commented on src/main/java/net/honux/chess/Board.java:
```diff
@@ -3,20 +3,82 @@
+        return whitePawns.size()+blackPawns.size();
+    }
+
+    public Pawn findWhitePawn(int index) {return whitePawns.get(index);}
```
항상 코드 일관성을 위해서 코드 리포맷을 해주시는것을 권장합니다.

ksundong replied:
만약에 index가 범위를 벗어나는 경우에는 어떻게 해결해주어야 할까요?

malaheaven replied:
이 또한 이럴 경우 오류를 발생시켜서 오류 메세지를 보여주는 것으로 해결을 했는데, 더 좋은 방법이 있다면 말씀해주세요 ㅎㅎ

ksundong replied:
확인해보겠습니다~

ksundong commented on src/main/java/net/honux/chess/Board.java:
```diff
@@ -3,20 +3,82 @@
+    public void initialize() {
+        int boardSize = 8;
+
+        for (int i = 1; i <= boardSize; i++) {
+            whitePawnsAdd(new Pawn(Pawn.WHITE_COLOR,Pawn.WHITE_REPRESENTATION));
+            blackPawnsAdd(new Pawn(Pawn.BLACK_COLOR,Pawn.BLACK_REPRESENTATION));
+        }
```
```suggestion
        for (int i = 0; i < boardSize; i++) {
            whitePawnsAdd(new Pawn(Pawn.WHITE_COLOR,Pawn.WHITE_REPRESENTATION));
            blackPawnsAdd(new Pawn(Pawn.BLACK_COLOR,Pawn.BLACK_REPRESENTATION));
        }
```
저는 보통 index를 0부터 시작하는 방법을 선호합니다. 이 방법에 익숙해지시는게 좋을 것 같아요!

malaheaven replied:
넵넵 그러네요 ㅎㅎ index는 0 부터 시작하도록 변경하겠습니다 :)

ksundong commented on src/main/java/net/honux/chess/Board.java:
```diff
@@ -3,20 +3,82 @@
+
+    public String print(){
+        StringBuilder printResult = new StringBuilder();
+        String line = "........";
```
`line`이라는 이름 너무 단순하지 않나요? `emptyLIne` 정도만 되어도 더 직관적일 것 같아요.

malaheaven replied:
넵넵 직관적인 네이밍이 너무 어렵네요 :)

ksundong commented on src/main/java/net/honux/chess/Board.java:
```diff
@@ -3,20 +3,82 @@
+        StringBuilder printResult = new StringBuilder();
+        String line = "........";
+        char lineBreak = '\n';
+        printResult.append(line);
+        printResult.append(lineBreak);
+        printResult.append(getBlackPawnsResult());
+        printResult.append(lineBreak);
+        printResult.append(line);
+        printResult.append(lineBreak);
+        printResult.append(line);
+        printResult.append(lineBreak);
+        printResult.append(line);
+        printResult.append(lineBreak);
+        printResult.append(line);
+        printResult.append(lineBreak);
+        printResult.append(getWhitePawnsResult());
+        printResult.append(lineBreak);
+        printResult.append(line);
+        return printResult.toString();
```
```suggestion
        printResult.append(line).append(lineBreak);
        printResult.append(getBlackPawnsResult()).append(lineBreak);
        printResult.append(line).append(lineBreak);
        printResult.append(line).append(lineBreak);
        printResult.append(line).append(lineBreak);
        printResult.append(line).append(lineBreak);
        printResult.append(getWhitePawnsResult()).append(lineBreak);
        printResult.append(line).append(lineBreak);
        return printResult.toString();
```
이렇게 붙여줘도 되고, printResult를 아예 없애줘도 좋을 것 같아요.

ksundong replied:
이렇게 라인을 줄인 이유는 세로로 너무 길어도 가독성이 떨어지기 떄문이에요.

malaheaven replied:
emptyLine을 메소드로 빼고, 줄바꿈을 다른 OS에서도 사용가능 할 수 있도록` System.getProperty("line.separator");` 로 변경했습니다.
디온이 말씀주신것과 같이 변경해보았는데, 확실히 전보다는 가독성이 좋아진 것 같지만, 아직도 가독성이 그리 좋지 않은 것 같아서,.. 어떻게 변경해야 더 눈에 확 잘 들어올지 고민이 되네요

malaheaven replied:
변경후

``` java
 private String emptyLine(){
        String emptyLine = "........";
        return emptyLine;
    }

    public String boardPrint() {
        StringBuilder printResult = new StringBuilder();
        String lineBreak = System.getProperty("line.separator");
        printResult.append(emptyLine()).append(lineBreak);
        printResult.append(getBlackPawnsResult()).append(lineBreak);
        printResult.append(emptyLine()).append(lineBreak);
        printResult.append(emptyLine()).append(lineBreak);
        printResult.append(emptyLine()).append(lineBreak);
        printResult.append(emptyLine()).append(lineBreak);
        printResult.append(getWhitePawnsResult()).append(lineBreak);
        printResult.append(emptyLine());
        return printResult.toString();
    }
```

ksundong replied:
정답이 없는 문제라 ㅎㅎ emptyLine에 lineBreak를 더해서 리턴해줘도 좋겠네요!

ksundong commented on src/main/java/net/honux/chess/Board.java:
```diff
@@ -3,20 +3,82 @@
+        return blackResult.toString();
+    }
+
+    public String print(){
```
`print()` 말고 더 좋은 이름을 고민해봐주세요!

malaheaven replied:
`boardPrint()` 가 더 직관적일까요? 우선 boardPrint()로 변경했어요

ksundong replied:
음 getRepresentation도 좋을 것 같고, 아무튼 print보다는 낫군요.

ksundong commented on src/main/java/net/honux/chess/GameMain.java:
```diff
@@ -0,0 +1,26 @@
+        while (true){
+            String command = sc.nextLine();
+            if (command.equals("start")){
+                Board board = new Board();
```
Board의 생성시점이 `start`라면, start를 입력할 때 마다 Board가 생기게 됩니다. 과연 괜찮을까요?

malaheaven replied:
start가 시작 지점이라, Board객체를 생성하고자 했는데, 보드는 매번 생성해주는 게 아닌 한 객체로만 동작 할 수 있도록 start 밖으로 빼주고 initialize는 start지점에서 해주는 게 더 좋겠네요!

ksundong commented on src/main/java/net/honux/chess/GameMain.java:
```diff
@@ -0,0 +1,26 @@
+                board.initialize();
+                System.out.println(board.print());
+
+            }else if (command.equals("end")) {
+                System.out.println("Bye~");
+                break;
+            }
```
요 문장을 위에 뽑아내는 편이 더 깔끔할 것 같아요.

ksundong replied:
예를들어 
```java
if ("end".equals(command)) {
    System.out.println("Bye~");
    break;
}
if ("start".equals(command)) {
    // 로직
}
```

굳이 else if를 쓸 필요는 없어요.

malaheaven replied:
넵넵 그 편이 코드가 더욱 깔끔해지네요

ksundong commented on src/main/java/net/honux/chess/GameMain.java:
```diff
@@ -0,0 +1,26 @@
+
+public class GameMain {
+    public static void main(String[] args) {
+        Scanner sc = new Scanner(System.in);
```
이 부분 `close`까지 다좋지만, try-with-resources에 대해서 공부해보세요.
참고로, 비정상 종료되면 Scanner는 close 되지 않는 코드입니다.

malaheaven replied:
이 부분 적용했는데, 좋네요! 좋은 공부를 했습니다 ㅎㅎ

ksundong commented on src/test/java/net/honux/chess/BoardTest.java:
```diff
@@ -4,27 +4,56 @@
-        verifyBoard(new Pawn("black"), 2, 1);
+    @DisplayName("WhitePawns의 값 사이즈 index로 색상 값을 잘 가져오는지 확인")
+    void create_white(){
+        Pawn white = new Pawn(Pawn.WHITE_COLOR,Pawn.WHITE_REPRESENTATION);
```
뭔가 실수하기 좋아보이네요. WHITE_COLOR, BLACK_REPRESENTATION을 넣으면 어떡하죠?

ksundong replied:
이 부분은 다음 단계에 포함되어 있으니 다음 단계에 개선해보세요!

malaheaven replied:
헉..그러네요 문제가 되겠어요..! 😱 다음 단계에 포함되어 있는 내용이라니, 그럼 이 부분은 다음 단계에서 개선해보도록 하겠습니다 😃

ksundong commented on src/test/java/net/honux/chess/BoardTest.java:
```diff
@@ -4,27 +4,56 @@
+    void create_white(){
+        Pawn white = new Pawn(Pawn.WHITE_COLOR,Pawn.WHITE_REPRESENTATION);
+        board.whitePawnsAdd(white);
+        assertThat(board.size()).isEqualTo(1);
+        assertThat(board.findWhitePawn(0).getColor()).isEqualTo("white");
```
되도록이면 확인할 수 있는 필드들은 전부 확인해주는 편이 좋아보이는 코드입니다.
이것만 확인해서는 정확한지 증명할 수 없어보여요.

malaheaven replied:
넵넵 확인할 수 있는 필드들은 전부 확인할 수 있도록 개선하겠습니다 😄

ksundong commented on src/test/java/net/honux/chess/BoardTest.java:
```diff
@@ -4,27 +4,56 @@
+    @DisplayName("보드 초기화가 잘 print 되는지 확인")
+    void print(){
+        board.initialize();
+        System.out.println(board.print());
```
이 코드.. 정말 이렇게 테스트 하는게 맞을까요?
출력값을 출력해서 테스트하는 것이 **자동화된** 단위테스트의 목적이 맞을까요?
어떻게하면 출력값을 테스트 할 수 있을까요?

malaheaven replied:
그러네요.. 출력 값을 출력해서 테스트 하는 것은 테스트 목적에 맞지 않네요.
그래서 변경을 아래와 같이 진행하려고 하는데 print 출력물 테스트는 이렇게 진행해도 될까요? 😄 
``` java
 @Test
    @DisplayName("보드 초기화가 잘 print 되는지 확인")
    void print(){
        board.initialize();
        assertThat(board.boardPrint()).isEqualTo("........"+System.getProperty("line.separator")+
                                            "PPPPPPPP"+System.getProperty("line.separator")+
                                            "........"+System.getProperty("line.separator")+
                                            "........"+System.getProperty("line.separator")+
                                            "........"+System.getProperty("line.separator")+
                                            "........"+System.getProperty("line.separator")+
                                            "pppppppp"+System.getProperty("line.separator")+
                                            "........");
    }
```

ksundong replied:
네 하드코딩으로 원하는 바가 맞는지 확인할 수 있다면 적당합니다.

ksundong commented on src/test/java/net/honux/chess/pieces/PawnTest.java:
```diff
@@ -5,25 +5,26 @@
-        verifyPawn("black");
-
+    @DisplayName("기본 생성자는 흰색 말과 p로 표시한다")
+    void create_기본생성자() throws Exception {
```
한영 혼용은 별로 추천드리지 않습니다.

malaheaven replied:
넵넵 피드백 감사합니다:)

ksundong commented on src/test/java/net/honux/chess/pieces/PawnTest.java:
```diff
@@ -5,25 +5,26 @@
-        assertThat(pawn.getColor()).isEqualTo(color);
+    @Test
+    @DisplayName("해당 색상과 대표문자를 잘 가져오는지 확인한다")
+    void create() {
```
메서드명에도 신경을 많이 써주세요~

malaheaven replied:
넵넵 변경하겠습니다 :)

ksundong reviewed (changes requested):
안녕하세요. 리뷰어 디온입니다. 미션3까지 오느라 고생많으셨습니다.
커밋로그는 학습단계이기 때문에 조금 더 성실히 작성해주시면 좋을 것 같아요.
가급적이면 현재단계에서는 심하면 메서드 단위로 커밋을 해주셔도 좋을 것 같습니다.
그래도 커밋 로그를 자세히 적어주셔서 좋은 것 같아요. (다만 커밋 body는 읽을일이 잘 없어서... 되도록이면 제목이 직관적이게 적어주시는 편이 좋다고 생각합니다.)
"및" 같은 단어가 들어간다거나 "," 같은 부호가 들어가면 조금 신경써보시는걸 추천드려요.
제목은 너무 길면 짤리니까 적당하게 작성해주세요.
> Board 클래스 ArrayList<Pawn> pawns = new ArrayList<Pawn> 로 되어있는 부분 List<Pawn> pawns = new ArrayList<List>로 변경
의 경우에는 너무 길어요~

예제코드는 진행이 막힐 때만 참고하시고 되도록이면 멤버분들과 얘기도 나눠보시면서 다양한 의견을 들어보세요.

몇몇 변경하시면 좋을 부분이 있어 변경 요청드립니다. 보시면서 궁금하시거나, 잘 이해가 되지 않는 부분은 코멘트 남겨주시면 확인 후 답변드릴게요.

## 리뷰

- 요구사항 정리를 꼼꼼하게 잘 해주셨습니다. 좋은 습관이네요.
- 접근 제어자는 항상 유의해줍시다. 꼭 짚고 넘어가셨으면 좋겠어요.
- 예외 상황도 항상 염두에 두는 편이 좋습니다. 사실 눈에 잘 안들어 올거에요.
- 코드 리포맷을 습관화 해주세요. 코드 일관성을 위해서 꼭 필요합니다.
- for문 index는 직관적인 경우를 제외하고는 0부터 시작하는 습관을 추천합니다.
- 세로 방향으로 너무 길어도 가독성이 떨어지게 됩니다. 적절한 수준으로 만들어주세요.
- 네이밍도 항상 고민 후에 작성해주세요. 가장 어려운 부분이지만, 꼭 필요합니다.
- try-with-resources에 대해서 꼭 공부해보세요.
- Test Code는 단순히 print만 찍어선 안됩니다. assertion을 넣어주세요.

요구사항은 잘 구현해주신 것 같아서 변경 요청드린 부분만 좀 더 개선해주시면 될 것 같아요. 고생하셨습니다~

malaheaven reviewed (commented):
[금일 1차 피드백 검토]
@ksundong 꼼꼼한 코드리뷰 정말 감사합니다 😄 
제가 코멘트 단 부분은 리뷰 토대로 수정을 진행한 부분입니다.
코멘트를 달지 않은 부분은 수정을 진행할 예정으로, 천천히 수정하겠습니다 :)

ksundong commented:
@malaheaven 네 수정하시고 꼭 푸시 부탁드려요~
푸시하시고 리뷰 요청 다시보내주세요~

malaheaven committed:
리뷰 내용 반영

- List 접근 제어자
- whitePawnsAdd() -> addWhitePawns() 로 변경
- blackPawnsAdd() -> addBlackPawns() 로 변경
- initialize() for i 0부터 시작하도록 변경
- print() -> boardPrint()로 변경
- emptyLine() 메소드 생성

malaheaven committed:
리뷰 내용 반영 try-with-resources 내용 반영

비정상 종료되면 scanner는 close 되지 않는 코드 try-with-resources 사용해서 정상 close 되도록 변경

malaheaven committed:
리뷰 내용 반영

- 과거 print()에서 출력 테스트를 출력을 통해 진행했던 테스트 메소드 변경

malaheaven committed:
리뷰 내용 반영

malaheaven commented:
@ksundong 디온 내용 수정 완료 했습니다 리뷰 요청드립니다.

ksundong commented on src/main/java/net/honux/chess/Board.java:
```diff
@@ -1,22 +1,96 @@
 package net.honux.chess;
 
 import net.honux.chess.pieces.Pawn;
+import org.omg.CORBA.DynAnyPackage.InvalidValue;
```
음...  이 부분은 왜 들어왔을까요?

ksundong commented on src/main/java/net/honux/chess/Board.java:
```diff
@@ -1,22 +1,96 @@
 import net.honux.chess.pieces.Pawn;
+import org.omg.CORBA.DynAnyPackage.InvalidValue;
 
+import java.security.InvalidParameterException;
```
https://docs.oracle.com/javase/7/docs/api/java/security/InvalidParameterException.html
요 예외는 우리가 일반적으로 사용하는 예외가 아닙니다. 예외를 직접 정의해주는 방안도 고려해보세요!

ksundong commented on src/main/java/net/honux/chess/Board.java:
```diff
@@ -1,22 +1,96 @@
+    public void addWhitePawns(Pawn pawn) {
+        if (!pawn.getColor().equals(Pawn.WHITE_COLOR)) {
+            throw new InvalidParameterException("The color of this pawn must be white.");
+        }whitePawns.add(pawn);
```
```suggestion
        }
        whitePawns.add(pawn);
```

ksundong commented on src/main/java/net/honux/chess/GameMain.java:
```diff
@@ -0,0 +1,29 @@
+        System.out.println("> 체스 게임을 시작하시려면 start, 끝내시려면 end 명령을 입력하세요.");
+        Board board = new Board();
+
+        String command;
+        try (Scanner sc = new Scanner(System.in);) {
+            while (true) {
+                command = sc.nextLine();
+                if (command.equals("end")) {
+                    System.out.println("Bye~");
+                    break;
+                }
+
+                if (command.equals("start")) {
+                    board.initialize();
+                    System.out.println(board.boardPrint());
+                }
+            }
+        } catch (Exception e) {
+            System.out.println(e);
+        }
```
```suggestion
        try (Scanner sc = new Scanner(System.in)) {
            String command;
            while (true) {
                command = sc.nextLine();
                if (command.equals("end")) {
                    System.out.println("Bye~");
                    break;
                }

                if (command.equals("start")) {
                    board.initialize();
                    System.out.println(board.boardPrint());
                }
            }
        }
```

이렇게도 고칠 수 있을 것 같아보이는데 한 번 확인해주세요!

ksundong commented on src/test/java/net/honux/chess/BoardTest.java:
```diff
@@ -4,27 +4,68 @@
+    void create_white() {
+        Pawn white = new Pawn(Pawn.WHITE_COLOR, Pawn.WHITE_REPRESENTATION);
+        board.addWhitePawns(white);
+        assertThat(board.size()).isEqualTo(1);
+        assertThat(board.findWhitePawn(0)).isEqualTo(white);
+        assertThat(board.findWhitePawn(0).getColor()).isEqualTo("white");
+        assertThat(board.findWhitePawn(0).getRepresentation()).isEqualTo('p');
```
같은 의미단위의 assertion이라면 `assertAll()`도 한 번 고려해봐주세요!

ksundong commented on src/test/java/net/honux/chess/pieces/PawnTest.java:
```diff
@@ -5,25 +5,26 @@
-        verifyPawn("black");
-
+    @DisplayName("기본 생성자는 흰색 말과 p로 표시한다")
+    void create_default_constructor() throws Exception {
```
```suggestion
    void create_default_constructor() {
```

ksundong reviewed (approved):
Robin! 리뷰 남긴 내용 확인해주시고, 현재 단계에서 반영하실 내용은 반영해주세요.
그 다음 바로 다음단계 진행해주세요~ 고생하셨습니다!

malaheaven commented:
@ksundong 넵넵 디온 주말인데도 불구하고, 또 한번의 꼼꼼한 리뷰 감사합니다! 리뷰를 받아 너무 좋네요 👍

honux77 commented:
👍
